test/docker_container_test_case.py

```diff
@@ -81,22 +85,32 @@ def _create_container(cls, image: str, container_port: int, **kwargs) -> Netloc:
                                                ports=ports,
                                                **kwargs)
         try:
+            time_it = datetime.now()
             container_info = cls._docker.api.inspect_container(container.id)
-            network_settings = container_info['NetworkSettings']
             if is_sibling:  # no coverage
-                container_ip = network_settings['IPAddress']
+                container_ip = container_info['NetworkSettings']['IPAddress']
                 assert isinstance(container_ip, str)
                 endpoint = (container_ip, container_port)
                 log.info('Launched sibling container %s from image %s, listening on %s:%i',
                          container.name, image, container_ip, container_port)
             else:
-                ports = network_settings['Ports']
-                port = one(ports[f'{container_port}/tcp'])
+                seconds = 0.0
+                while True:
+                    ports = container_info['NetworkSettings']['Ports'][f'{container_port}/tcp']
```

Comment:
```suggestion
                while True:
                    seconds = (datetime.now() - time_it).total_seconds()
                    ports = container_info['NetworkSettings']['Ports'][f'{container_port}/tcp']
```

Moving the time calculation to the start of the while loop will let `seconds` reflect the total time taken even if sleep() is not called.